src/Tribe/Integrations/WPML/Linked_Posts.php

```diff
@@ -0,0 +1,152 @@
+<?php
+/**
+ * Class Tribe__Events__Integrations__WPML__Linked_Posts
+ *
+ * Handles linked posts fetching taking WPML managed translations into account.
+ */
+class Tribe__Events__Integrations__WPML__Linked_Posts {
+
+	/**
+	 * @var Tribe__Events__Integrations__WPML__Linked_Posts
+	 */
+	protected static $instance;
+
+	/**
+	 * @return Tribe__Events__Integrations__WPML__Linked_Posts
+	 */
+	public static function instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Assign linked posts managed by The Events Calendar a language.
+	 *
+	 * We use the filter as an action to assign linked posts a language.
+	 * WPML will not "see" posts that have not a language assigned: here we make sure that linked posts like
+	 * venues and organizers will be assigned the language of the event they are being linked to.
+	 *
+	 * @param int    $id               The linked post ID; this would be `null` by default but we know TEC is inserting
+	 *                                 the post at priority 10.
+	 * @param array  $data             An array of data representing the linked post submission.
+	 * @param string $linked_post_type The linked post type, e.g. `tribe_venue` or `tribe_organizer`.
+	 * @param string $post_status      The linked post type post status.
+	 * @param int    $event_id         The post ID of the event this post is linked to; this will be null for newly created events.
+	 *
+	 * @return int The untouched linked post ID.
+	 */
+	public function filter_tribe_events_linked_post_create( $id, $data, $linked_post_type, $post_status, $event_id ) {
```

Comment:
Right now, we don't seem to do anything with `$data` or `$post_status` - are they needed/are they being kept for future use? If the latter, _might_ be worth noting as much in the @param descriptions.